package/MDAnalysis/analysis/rms.py

```diff
@@ -574,56 +575,46 @@ def run(self, start=None, stop=None, step=None, progout=10,
             which means that the trajectory would be read until the end.
         step : int (optional)
             step between frames, default None becomes 1.
-        progout : int (optional)
-            number of frames to iterate through between updates to progress
-            output; ``None`` for no updates [10]
+        weights : str / array_like (optional)
+            used weights. If ``'mass'`` use masses of atomgroup, it ``None`` use uniform weights.
         verbose : bool (optional)
-            if ``False``, suppress all output (implies *progout* = ``None``)
-            [``True``]
-
-        References
-        ----------
-        .. [Welford1962] B. P. Welford (1962). "Note on a Method for
-           Calculating Corrected Sums of Squares and Products." Technometrics
-           4(3):419-420.
-
-        .. deprecated:: 0.16
-           The keyword argument *quiet* is deprecated in favor of *verbose*.
+            if ``False``, suppress all output
         """
-        traj = self.atomgroup.universe.trajectory
-        start, stop, step = traj.check_slice_indices(start, stop, step)
-        sumsquares = np.zeros((self.atomgroup.n_atoms, 3))
-        means = np.array(sumsquares)
-
-        verbose = _set_verbose(verbose, quiet, default=True)
-        if not verbose:
-            progout = None
-
-        # set up progress output
-        if progout:
-            percentage = ProgressMeter(self.atomgroup.universe.trajectory.n_frames,
-                                       interval=progout)
-        else:
-            percentage = ProgressMeter(self.atomgroup.universe.trajectory.n_frames,
-                                       verbose=False)
+        super(RMSF, self).__init__(atomgroup.universe.trajectory, **kwargs)
+        self.atomgroup = atomgroup
+        if weights == 'mass':
+            weights = self.atomgroup.masses
+        self.weights = weights
+
+    def run(self, start=None, stop=None, step=None, progout=None,
```

Comment:
@jbarnoud This allows to use the class also in the old style and it prints a warnings.